src/main/java/io/neonbee/config/CorsConfig.java

```diff
@@ -0,0 +1,271 @@
+        return this;
+    }
+
+    /**
+     * Returns the set of allowed headers.
+     * <p>
+     * By default, this is null, which means the header <i>access-control-allow-headers</i> is not sent at all.
+     *
+     * @return the set of allowed headers, or null íf no header is set.
+     */
+    public Set<String> getAllowedHeaders() {
+        return allowedHeaders;
+    }
+
+    /**
+     * Set a set of allowed headers.
+     *
+     * @param allowedHeaders the allowed header names
+     * @return the {@linkplain CorsConfig} for fluent use
+     */
+    @Fluent
+    public CorsConfig setAllowedHeaders(Set<String> allowedHeaders) {
+        this.allowedHeaders = allowedHeaders;
+        return this;
+    }
+
+    /**
+     * Returns the set of exposed headers.
+     * <p>
+     * By default, this is null, which means the header <i>access-control-expose-headers</i> is not sent at all.
+     *
+     * @return the set of allowed headers, or null íf no exposed header is set.
+     */
+    public Set<String> getExposedHeaders() {
+        return exposedHeaders;
+    }
+
+    /**
+     * Set a set of exposed headers.
+     *
+     * @param exposedHeaders the exposed header names
+     * @return the {@linkplain CorsConfig} for fluent use
+     */
+    @Fluent
+    public CorsConfig setExposedHeaders(Set<String> exposedHeaders) {
+        this.exposedHeaders = exposedHeaders;
+        return this;
+    }
+
+    /**
+     * Returns how long the browser should cache the information in seconds.
+     * <p>
+     * By default, it is -1, which means the header <i>access-control-max-age</i> is not sent at all.
+     *
+     * @return how long the browser should cache the information.
+     */
+    public int getMaxAgeSeconds() {
+        return maxAgeSeconds;
+    }
+
+    /**
+     * Set how long the browser should cache the information.
+     *
+     * @param maxAgeSeconds max age in seconds
+     * @return the {@linkplain CorsConfig} for fluent use
+     */
+    @Fluent
+    public CorsConfig setMaxAgeSeconds(int maxAgeSeconds) {
+        this.maxAgeSeconds = maxAgeSeconds;
+        return this;
+    }
+
+    /**
+     * Are credentials are allowed?
+     * <p>
+     * By default, it is not allowed.
+     *
+     * @return true if credentials are allowed, otherwise false.
+     */
+    @SuppressWarnings("PMD.BooleanGetMethodName")
+    public boolean getAllowCredentials() {
+        return allowCredentials;
+    }
+
+    /**
+     * Set whether credentials are allowed or not.
+     *
+     * @param allowCredentials true if allowed
+     * @return the {@linkplain CorsConfig} for fluent use
+     */
+    @Fluent
+    public CorsConfig setAllowCredentials(boolean allowCredentials) {
+        this.allowCredentials = allowCredentials;
+        return this;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof CorsConfig)) {
```

Comment:
I do not expect subclasses of CorsConfig. So wouldn't  "... getClass() != o.getClass() ... " be the more appropriate approach, because only the exact type is important?

Reply:
Now the voter fails with:

```
/home/runner/work/neonbee/neonbee/src/main/java/io/neonbee/config/CorsConfig.java:250: warning: [EqualsGetClass] Prefer instanceof to getClass when implementing Object#equals.
```

So I will change it back